src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -235,6 +213,36 @@ def _parse_image_argument(namespace):
         raise CLIError(err.format(namespace.image, [x['urnAlias'] for x in images]))
 
 
+def _get_image_plan_info_if_exists(namespace):
+    try:
+        compute_client = _compute_client_factory()
+        if namespace.os_version.lower() == 'latest':
+            top_one = compute_client.virtual_machine_images.list(namespace.location,
```

Comment:
Please verify if the `top_one` instance has `plan` property. If so you can return the value directly here without fetching the image again on line 233.

Reply:
Verified that the `list` still returns `VirtualMachineImageResource` which doesn't have the `plan` information, so we have to invoke a `get`